def/es-proposals.js

```diff
@@ -0,0 +1,33 @@
+module.exports = function (fork) {
+  fork.use(require('./core'));
+
+  var types = fork.use(require("../lib/types"));
+  var Type = types.Type;
+  var def = types.Type.def;
+  var or = Type.or;
+
+  var shared = fork.use(require("../lib/shared"));
+  var defaults = shared.defaults;
+
+
+  // https://github.com/tc39/proposal-optional-chaining
+  // `a?.b` as per https://github.com/estree/estree/issues/146
+  def("OptionalMemberExpression")
+    .bases("MemberExpression")
+    .build("object", "property", "computed", "optional")
+    .field("optional", Boolean, defaults["true"])
+
+  // a?.b()
+  def("OptionalCallExpression")
```

Comment:
How is this different from a `CallExpression` whose `callee` is an `OptionalMemberExpression`? Or can you do something like `f?()`?

Reply:
No, this is more like `a?.b()`. Since the operator is short circuiting the AST needs to know whether the method call is optional as well.

Reply:
Actually, I double checked this and I stand corrected. Yes, it's `a?.()`.
See https://github.com/tc39/proposal-optional-chaining#base-case